source/ui/util/AMDialog.cpp

```diff
@@ -37,6 +41,24 @@ void AMDialog::layoutDialogButtons()
 	buttonsLayout_->addWidget(buttonBox);
 }
 
+void AMDialog::setEnableDialogEnterKeyBehavior(bool enable)
+{
+	enableDefaultEnterBehavior_ = enable;
+}
+
+void AMDialog::keyPressEvent(QKeyEvent *e)
+{
+	switch (e->key ()) {
+	case Qt::Key_Return:
+	case Qt::Key_Enter:
```

Comment:
I don't think this is the way you want to use this.  This negates any use of the enter key not just closing the dialog.  What you likely need to do is change the default key of the dialog to not trigger the accept() flag.